### PR TITLE
php: Update extensions and packages

### PR DESCRIPTION
**Pull request by drupol**
###### Motivation for this change

This PR:
- [x] Provide the `src` attribute using `fetchFromGitHub` of some packages and/or extensions, so it can be picked up by the r-ryantm bot and provide PR to update version, right now, it doesn't.
- [x] Update the version of some packages and extensions so they are up-to-date.
- [x] Add missing calls to `preInstall` and `postInstall` hooks in `installPhase`
- [x] Sort and update metadata informations

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by drupol** [comment]
@GrahamcOfBorg build phpExtensions.amqp
@GrahamcOfBorg build phpExtensions.apcu
@GrahamcOfBorg build phpExtensions.apcu_bc
@GrahamcOfBorg build phpExtensions.ast
@GrahamcOfBorg build phpPackages.composer
@GrahamcOfBorg build phpExtensions.ds
@GrahamcOfBorg build phpExtensions.pcov
@GrahamcOfBorg build phpPackages.php-cs-fixer
@GrahamcOfBorg build phpPackages.php-parallel-lint
@GrahamcOfBorg build phpPackages.phpcs
@GrahamcOfBorg build phpPackages.phpmd
@GrahamcOfBorg build phpPackages.phpstan
@GrahamcOfBorg build phpPackages.psysh
@GrahamcOfBorg build phpExtensions.redis
@GrahamcOfBorg build phpExtensions.xdebug

**Comment by drupol** [comment]
@GrahamcOfBorg build phpExtensions.amqp
@GrahamcOfBorg build phpExtensions.apcu
@GrahamcOfBorg build phpExtensions.apcu_bc
@GrahamcOfBorg build phpExtensions.ast
@GrahamcOfBorg build phpPackages.composer
@GrahamcOfBorg build phpExtensions.ds
@GrahamcOfBorg build phpExtensions.pcov
@GrahamcOfBorg build phpPackages.php-cs-fixer
@GrahamcOfBorg build phpPackages.php-parallel-lint
@GrahamcOfBorg build phpPackages.phpcs
@GrahamcOfBorg build phpPackages.phpmd
@GrahamcOfBorg build phpPackages.phpstan
@GrahamcOfBorg build phpPackages.psysh
@GrahamcOfBorg build phpExtensions.redis
@GrahamcOfBorg build phpExtensions.xdebug

**Comment by drupol** [comment]
@GrahamcOfBorg build phpExtensions.amqp
@GrahamcOfBorg build phpExtensions.apcu
@GrahamcOfBorg build phpExtensions.apcu_bc
@GrahamcOfBorg build phpExtensions.ast
@GrahamcOfBorg build phpPackages.composer
@GrahamcOfBorg build phpExtensions.ds
@GrahamcOfBorg build phpExtensions.pcov
@GrahamcOfBorg build phpPackages.php-cs-fixer
@GrahamcOfBorg build phpPackages.php-parallel-lint
@GrahamcOfBorg build phpPackages.phpcs
@GrahamcOfBorg build phpPackages.phpmd
@GrahamcOfBorg build phpPackages.phpstan
@GrahamcOfBorg build phpPackages.psysh
@GrahamcOfBorg build phpExtensions.redis
@GrahamcOfBorg build phpExtensions.xdebug

**Comment by etu** [comment]
Hmm, I think two things :)

1. We should probably change the declaration of buildPecl as well to expect that input and if it doesn't exist -> fall back to previous behavior (this should be fairly easy to do)
2. I think if/when we do such a change, we should do it without the package version changes. Because if we do it without changes of versions/hashes and made correctly, we should be able to get it to zero (or very close to zero) rebuilds in the builder bot to easily confirm that it doesn't have any side-effects :)

**Review by aanderse — APPROVED** [review]
Having the bot automatically update these extensions would be a **big** win...

Any objections from @talyz or @Ma27? Let's give them a few days to respond and then merge?

**Review by etu — CHANGES_REQUESTED** [review]
In general I approve with the changes, but the commit history is a bit of a mess.

`php.extensions.composer: 2.2.3 -> 2.2.5` should be `php.packages.composer: 2.2.3 -> 2.2.5`

Same for:
 - php-cs-fixer
 - php-parallel-lint
 - phpcs
 - phpmd
 - phpstan
 - psalm
 - psysh

`amqp` and `redis` however belongs in `php.extensions`.

The rest looks fine, including the commit that refactors a bunch of extensions source attributes 🙂

**Comment by drupol** [comment]
Going to fix this asap.

File: pkgs/development/php-packages/php-parallel-lint/default.nix
Comment (drupol) [reply pkgs/development/php-packages/php-parallel-lint/default.nix:16]: I mostly do not agree and I will make my point, perhaps this is something I don't understand yet.

First, I don't think that these PHP projects should be packaged in Nix. I was surprised to see that they were available at first.
Why? They are available through `composer` and as a `phar`.

Second, let's say that we decide to bundle this package in Nix, then I don't think it's up to us to build it.
There is a phar provided by the vendor, we should use it.

Third, the issue you're mentioning is in your project, not in Nix, so, maybe we should mitigate and fix the issue you had in your package.

Last but not least, in order to unify and simplify things, it's always nice to have the same building procedure for all projects.
Then, if we decide to build this PHP project, then we should also decide to do it for all the others that are in nixpkgs (phpstan, psalm, etc etc).

Anyway. I'm in this community since not so long ago so my voice would not weight in the balance, but this is what I think as a PHP developer.

File: pkgs/development/php-packages/amqp/default.nix
Comment (drupol) [reply pkgs/development/php-packages/amqp/default.nix:7]: I'm not yet used to `rec`. Is there an issue doing what I did or `rec` is just another syntax for the same thing?

File: pkgs/development/php-packages/php-parallel-lint/default.nix
Comment (jtojnar) [reply pkgs/development/php-packages/php-parallel-lint/default.nix:16]: > First, I don't think that these PHP projects should be packaged in Nix. I was surprised to see that they were available at first. Why? They are available through `composer` and as a `phar`.

- Nix has much better ergonomics than composer:
  - there is no `nix-shell` equivalent
  - depending on who you ask, you are supposed to install projects that you do not want to or cannot (e.g. due to conflicting dependencies) install into your project using `composer create-project` or `composer global require`, which are both broken.
- phars are better but how are you supposed to use them?
  - downloading manually makes updating cumbersome
  - what is even [phive](https://phar.io/)?
  - Nix is the least bad package manager there is so we might as well use it
 
> Second, let's say that we decide to bundle this package in Nix, then I don't think it's up to us to build it. There is a phar provided by the vendor, we should use it.

As you noticed, there are two somewhat conflicting goals. Nix is source-based package manager and we prefer building packages from source where practical, since it allow us to fix various issues trivially (bugs that need patching, dependencies that need updating). But we also want to have the packages as close to upstream as possible to minimize the risk of weirdness. If we took the latter to conclusion, we would only use binary packages provided by upstream. But that would go against the first goal, which is IMO more important for a Linux distribution.

> Third, the issue you're mentioning is in your project, not in Nix, so, maybe we should mitigate and fix the issue you had in your package.

Yes, in the end it turned out that it was issue in our PHP packaging but workaround would still require rebuilding the phar ourselves. Issues that require patching the source or tweaking the build process are bread and butter of package maintainer. 

> Last but not least, in order to unify and simplify things, it's always nice to have the same building procedure for all projects. Then, if we decide to build this PHP project, then we should also decide to do it for all the others that are in nixpkgs (phpstan, psalm, etc etc).

Yes unification is a nice thing to strive for but different packages have different maintainers and it is not really fair to expect a person who wants to have a package available to also normalize all other packages they might not even use.

> Anyway. I'm in this community since not so long ago so my voice would not weight in the balance, but this is what I think as a PHP developer.

I think fresh perspective is always useful since we might accustomed to the status quo without realizing that it is suboptimal. But it is also important to realize that each custom usually has a reason behind it.

File: pkgs/development/php-packages/amqp/default.nix
Comment (jtojnar) [reply pkgs/development/php-packages/amqp/default.nix:7]: Recursive attrsets allow you to refer to other attributes in the set. It is mostly the same, only slightly more concise:

```suggestion
buildPecl rec {
  pname = "amqp";
  version = "1.11.0";
```

Comment (drupol) [reply pkgs/development/php-packages/amqp/default.nix:7]: I though that if we could avoid `rec`, we should do it.

Anyway, I'll follow your advice and update my changes to reflect this.

File: pkgs/development/php-packages/php-parallel-lint/default.nix
Comment (drupol) [reply pkgs/development/php-packages/php-parallel-lint/default.nix:16]: >     * Nix has much better ergonomics than composer:
>       
>       * there is no `nix-shell` equivalent
>       * depending on who you ask, you are supposed to install projects that you do not want to or cannot (e.g. due to conflicting dependencies) install into your project using `composer create-project` or `composer global require`, which are both broken.
> 
>     * phars are better but how are you supposed to use them?
>       
>       * downloading manually makes updating cumbersome
>       * what is even [phive](https://phar.io/)?
>       * Nix is the least bad package manager there is so we might as well use it

I'm not comparing `composer` and Nix. Those are 2 completely different things for me, no discussion.
What I meant is that, when I need `php-parallel-lint` in [a project](https://packagist.org/packages/drupol/php-conventions), I add it myself in `require-dev` and do not rely on the underlying OS to provide it for me. And I still believe that these packages does not have their place in Nix... but that's another subject.

"phars are better" - let me explain.
composer, phpstan, psalm... are all providing PHARS that are built out of sources in Github Actions.
Millions of people are relying on these phars already and there unlikely few chances that it would broke one day.
Even the default `composer` [installer](https://getcomposer.org/download/) is using the phar version.

Another example, when PHP developers are requiring `phpstan/phpstan` in any projects, they are actually downloading the PHAR version of the project, not the PHP sources. This has the advantages of making sure that it works and there won't be any dependency conflicts. The phar version of PHPStan embed namespaced dependencies only required by PHPStan and does not interfere with the local project. The PHAR building is completely automatized on Github actions as well.

> As you noticed, there are two somewhat conflicting goals. Nix is source-based package manager and we prefer building packages from source where practical, since it allow us to fix various issues trivially (bugs that need patching, dependencies that need updating). But we also want to have the packages as close to upstream as possible to minimize the risk of weirdness. If we took the latter to conclusion, we would only use binary packages provided by upstream. But that would go against the first goal, which is IMO more important for a Linux distribution.

I understand. Then maybe in that case, we should provide 2 packages ?
* `php-parallel-lint` would be the version compiled from sources
* `php-parallel-lint-bin` would be the PHAR version.

I'm all in favor of open-source but I have the feeling that we are redoing twice(or more) the job here and this is what I'm trying to avoid at all costs (_for many reasons that I would not expose here but mostly environmental_). If the vendor is providing the PHAR, automatically built on Github actions, we should use it. That's usually the default way of using such packages anyway.

> Yes unification is a nice thing to strive for but different packages have different maintainers and it is not really fair to expect a person who wants to have a package available to also normalize all other packages they might not even use.

About `php-parallel-lint`, I think it worth to mention that I'm updating this package from version 1.0.0 (2018) to version 1.3.1 (2021) and things has changed a lot. Maybe we should try to use the PHAR for the time being so it's aligned with the rest of phar packages in use, and see how it behaves? Let's give it a try and as soon as there is an issue, I'll be the first to submit the PR to built this out of the source.

> I think fresh perspective is always useful since we might accustomed to the status quo without realizing that it is suboptimal. But it is also important to realize that each custom usually has a reason behind it.

Thanks, I was a bit afraid that this message was misunderstood, and I'm happy to see that it is a constructive discussion in the end, let's keep it like that.

File: pkgs/development/php-packages/amqp/default.nix
Comment (drupol) [reply pkgs/development/php-packages/amqp/default.nix]: Job done.

File: pkgs/development/php-packages/php-parallel-lint/default.nix
Comment (jtojnar) [reply pkgs/development/php-packages/php-parallel-lint/default.nix:16]: > I'm not comparing `composer` and Nix. Those are 2 completely different things for me, no discussion. What I meant is that, when I need `php-parallel-lint` in [a project](https://packagist.org/packages/drupol/php-conventions), I add it myself in `require-dev` and do not rely on the underlying OS to provide it for me. And I still believe that these packages does not have their place in Nix... but that's another subject.

Well sometimes you cannot add a project to `require-dev` (for example because of conflicting dependencies) or you do not want to do that (I regularly run `nix shell -f /home/jtojnar/Projects/nix-phps outputs.packages.x86_64-linux.php56 outputs.packages.x86_64-linux.php56.packages.php-parallel-lint -c php-parallel-lint src` when contributing to third party libraries.)

> "phars are better" - let me explain. composer, phpstan, psalm... are all providing PHARS that are built out of sources in Github Actions. Millions of people are relying on these phars already and there unlikely few chances that it would broke one day. Even the default `composer` [installer](https://getcomposer.org/download/) is using the phar version.

Well, you could say the same thing for binaries. Though it is true that this is a less of an issue here because PHP projects’ build process is less complex and there are no ABI boundaries.

> Another example, when PHP developers are requiring `phpstan/phpstan` in any projects, they are actually downloading the PHAR version of the project, not the PHP sources. This has the advantages of making sure that it works and there won't be any dependency conflicts. The phar version of PHPStan embed namespaced dependencies only required by PHPStan and does not interfere with the local project. The PHAR building is completely automatized on Github actions as well.

I am aware of that but that is still pretty rare for Packagist packages. And not without downsides either – particularly, it makes debugging issues with the tool hard.

> > As you noticed, there are two somewhat conflicting goals. Nix is source-based package manager and we prefer building packages from source where practical, since it allow us to fix various issues trivially (bugs that need patching, dependencies that need updating). But we also want to have the packages as close to upstream as possible to minimize the risk of weirdness. If we took the latter to conclusion, we would only use binary packages provided by upstream. But that would go against the first goal, which is IMO more important for a Linux distribution.
> 
> I understand. Then maybe in that case, we should provide 2 packages ?
> 
>     * `php-parallel-lint` would be the version compiled from sources
> 
>     * `php-parallel-lint-bin` would be the PHAR version.

Providing both versions is also a possibility but it brings costs of its own – now consumers will need to decide which version they want to use. So it IMO only makes sense when the two versions differ significantly (e.g. `firefox-bin` is more performant but `firefox` supports declarative customization).

> I'm all in favor of open-source but I have the feeling that we are redoing twice(or more) the job here and this is what I'm trying to avoid at all costs (_for many reasons that I would not expose here but mostly environmental_). If the vendor is providing the PHAR, automatically built on Github actions, we should use it. That's usually the default way of using such packages anyway.

Well, that is precisely what I am not sure about. The current Nixpkgs policy is to build stuff from source unless too inconvenient. I would say the primary reason most projects used PHARs from upstream instead of building them from scratch is that it is not easy to do without IFD. But we should strive to respect the policy when it is not a great PITA. Particularly, php-parallel lint has no other runtime dependencies than PHP itself.

The environmental impact will be almost certainly overshadowed by every other part of the build since building phar is mostly just a compression.

Comment (drupol) [reply pkgs/development/php-packages/php-parallel-lint/default.nix]: Thanks for taking the time to carefully explain each points.

I still do not agree with some of them, but I'm here to learn and I accept our disagreement.
The goal here is to move forward anyway :)

I did the changes you suggest and I hope it's ok now.

**Comment by drupol** [comment]
@GrahamcOfBorg build phpExtensions.amqp
@GrahamcOfBorg build phpExtensions.apcu
@GrahamcOfBorg build phpExtensions.apcu_bc
@GrahamcOfBorg build phpExtensions.ast
@GrahamcOfBorg build phpPackages.composer
@GrahamcOfBorg build phpExtensions.ds
@GrahamcOfBorg build phpExtensions.pcov
@GrahamcOfBorg build phpPackages.php-cs-fixer
@GrahamcOfBorg build phpPackages.php-parallel-lint
@GrahamcOfBorg build phpPackages.phpcs
@GrahamcOfBorg build phpPackages.phpmd
@GrahamcOfBorg build phpPackages.phpstan
@GrahamcOfBorg build phpPackages.psysh
@GrahamcOfBorg build phpExtensions.redis
@GrahamcOfBorg build phpExtensions.xdebug

**Comment by drupol** [comment]
@GrahamcOfBorg build phpExtensions.amqp
@GrahamcOfBorg build phpExtensions.apcu
@GrahamcOfBorg build phpExtensions.apcu_bc
@GrahamcOfBorg build phpExtensions.ast
@GrahamcOfBorg build phpPackages.composer
@GrahamcOfBorg build phpExtensions.ds
@GrahamcOfBorg build phpExtensions.pcov
@GrahamcOfBorg build phpPackages.php-cs-fixer
@GrahamcOfBorg build phpPackages.php-parallel-lint
@GrahamcOfBorg build phpPackages.phpcs
@GrahamcOfBorg build phpPackages.phpmd
@GrahamcOfBorg build phpPackages.phpstan
@GrahamcOfBorg build phpPackages.psysh
@GrahamcOfBorg build phpExtensions.redis
@GrahamcOfBorg build phpExtensions.xdebug

**Comment by drupol** [comment]
@GrahamcOfBorg build phpExtensions.amqp
@GrahamcOfBorg build phpExtensions.apcu
@GrahamcOfBorg build phpExtensions.apcu_bc
@GrahamcOfBorg build phpExtensions.ast
@GrahamcOfBorg build phpPackages.composer
@GrahamcOfBorg build phpExtensions.ds
@GrahamcOfBorg build phpExtensions.pcov
@GrahamcOfBorg build phpPackages.php-cs-fixer
@GrahamcOfBorg build phpPackages.php-parallel-lint
@GrahamcOfBorg build phpPackages.phpcs
@GrahamcOfBorg build phpPackages.phpmd
@GrahamcOfBorg build phpPackages.phpstan
@GrahamcOfBorg build phpPackages.psysh
@GrahamcOfBorg build phpExtensions.redis
@GrahamcOfBorg build phpExtensions.xdebug

**Comment by drupol** [comment]
@GrahamcOfBorg build phpExtensions.amqp
@GrahamcOfBorg build phpExtensions.apcu
@GrahamcOfBorg build phpExtensions.apcu_bc
@GrahamcOfBorg build phpExtensions.ast
@GrahamcOfBorg build phpPackages.composer
@GrahamcOfBorg build phpExtensions.ds
@GrahamcOfBorg build phpExtensions.pcov
@GrahamcOfBorg build phpPackages.php-cs-fixer
@GrahamcOfBorg build phpPackages.php-parallel-lint
@GrahamcOfBorg build phpPackages.phpcs
@GrahamcOfBorg build phpPackages.phpmd
@GrahamcOfBorg build phpPackages.phpstan
@GrahamcOfBorg build phpPackages.psalm
@GrahamcOfBorg build phpPackages.psysh
@GrahamcOfBorg build phpExtensions.redis
@GrahamcOfBorg build phpExtensions.xdebug

**Comment by drupol** [comment]
@GrahamcOfBorg build phpExtensions.amqp
@GrahamcOfBorg build phpExtensions.apcu
@GrahamcOfBorg build phpExtensions.apcu_bc
@GrahamcOfBorg build phpExtensions.ast
@GrahamcOfBorg build phpPackages.composer
@GrahamcOfBorg build phpExtensions.ds
@GrahamcOfBorg build phpExtensions.pcov
@GrahamcOfBorg build phpPackages.php-cs-fixer
@GrahamcOfBorg build phpPackages.php-parallel-lint
@GrahamcOfBorg build phpPackages.phpcs
@GrahamcOfBorg build phpPackages.phpmd
@GrahamcOfBorg build phpPackages.phpstan
@GrahamcOfBorg build phpPackages.psalm
@GrahamcOfBorg build phpPackages.psysh
@GrahamcOfBorg build phpExtensions.redis
@GrahamcOfBorg build phpExtensions.xdebug

**Comment by drupol** [comment]
@GrahamcOfBorg build phpExtensions.amqp
@GrahamcOfBorg build phpExtensions.apcu
@GrahamcOfBorg build phpExtensions.apcu_bc
@GrahamcOfBorg build phpExtensions.ast
@GrahamcOfBorg build phpPackages.composer
@GrahamcOfBorg build phpExtensions.ds
@GrahamcOfBorg build phpExtensions.pcov
@GrahamcOfBorg build phpPackages.php-cs-fixer
@GrahamcOfBorg build phpPackages.php-parallel-lint
@GrahamcOfBorg build phpPackages.phpcs
@GrahamcOfBorg build phpPackages.phpmd
@GrahamcOfBorg build phpPackages.phpstan
@GrahamcOfBorg build phpPackages.psalm
@GrahamcOfBorg build phpPackages.psysh
@GrahamcOfBorg build phpExtensions.redis
@GrahamcOfBorg build phpExtensions.xdebug

**Review by jtojnar — COMMENTED** [review]
I still feel somewhat unsure about fetching pecl packages from github, then the `buildPecl` name does not make much sense. But it is not that important and I will defer to the PHP maintainers. Otherwise it looks good.

**Comment by drupol** [comment]
Going to merge conflicts asap.

**Comment by drupol** [comment]
@GrahamcOfBorg build phpExtensions.amqp
@GrahamcOfBorg build phpExtensions.apcu
@GrahamcOfBorg build phpExtensions.apcu_bc
@GrahamcOfBorg build phpExtensions.ast
@GrahamcOfBorg build phpPackages.composer
@GrahamcOfBorg build phpExtensions.ds
@GrahamcOfBorg build phpExtensions.pcov
@GrahamcOfBorg build phpPackages.php-cs-fixer
@GrahamcOfBorg build phpPackages.php-parallel-lint
@GrahamcOfBorg build phpPackages.phpcs
@GrahamcOfBorg build phpPackages.phpmd
@GrahamcOfBorg build phpPackages.phpstan
@GrahamcOfBorg build phpPackages.psalm
@GrahamcOfBorg build phpPackages.psysh
@GrahamcOfBorg build phpExtensions.redis
@GrahamcOfBorg build phpExtensions.xdebug

**Review by etu — APPROVED** [review]

**Review by caugner — CHANGES_REQUESTED** [review]
I'm not against the changes, but I would like to see a more descriptive PR title and the "Motivation for change" section filled out.

As far as I can see, this PR not only updates extensions and packages, but it also changes the sources of some extensions and packages. IMO this should be clear from the PR title and mentioned in the "Motivation for change", especially as it was pointed out that changing the source may have side-effects.

**Comment by drupol** [comment]
> I'm not against the changes, but I would like to see a more descriptive PR title and the "Motivation for change" section filled out.
> 
> As far as I can see, this PR not only updates extensions and packages, but it also changes the sources of some extensions and packages. IMO this should be clear from the PR title and mentioned in the "Motivation for change", especially as it was pointed out that changing the source may have side-effects.

Hi !

I added the requested informations.

**Review by caugner — APPROVED** [review]

File: pkgs/development/php-packages/redis/default.nix
Comment (SuperSandro2000) [inline pkgs/development/php-packages/redis/default.nix:9]:
```suggestion
    rev = version;
```

File: pkgs/development/php-packages/redis/default.nix
Comment (SuperSandro2000) [inline pkgs/development/php-packages/redis/default.nix:2]:
```suggestion
{ buildPecl, lib, php, fetchFromGitHub }:

buildPecl rec {
```

Please keep the empty line here.

**Comment by drupol** [comment]
@GrahamcOfBorg build php.extensions.amqp php.extensions.apcu php.extensions.apcu_bc php.extensions.ast php.packages.composer php.extensions.ds php.extensions.pcov php.packages.php-cs-fixer php.packages.php-parallel-lint php.packages.phpcbf php.packages.phpcs php.packages.phpmd php.packages.phpstan php.packages.psalm php.packages.psysh php.extensions.redis php.extensions.xdebug

**Comment by drupol** [comment]
@GrahamcOfBorg build php.extensions.amqp php.extensions.apcu php.extensions.apcu_bc php.extensions.ast php.packages.composer php.extensions.ds php.extensions.pcov php.packages.php-cs-fixer php.packages.php-parallel-lint php.packages.phpcbf php.packages.phpcs php.packages.phpmd php.packages.phpstan php.packages.psalm php.packages.psysh php.extensions.redis php.extensions.xdebug

**Comment by drupol** [comment]
@GrahamcOfBorg build php.extensions.amqp php.extensions.apcu php.extensions.apcu_bc php.extensions.ast php.packages.composer php.extensions.ds php.extensions.pcov php.packages.php-cs-fixer php.packages.php-parallel-lint php.packages.phpcbf php.packages.phpcs php.packages.phpmd php.packages.phpstan php.packages.psalm php.packages.psysh php.extensions.redis php.extensions.xdebug

File: pkgs/development/php-packages/composer/default.nix
Comment (SuperSandro2000) [inline pkgs/development/php-packages/composer/default.nix:8]:
```suggestion
    url = "https://github.com/composer/composer/releases/download/${version}/composer.phar";
```
to easily get to the repo

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/development/php-packages/composer/default.nix
Comment (talyz) [reply pkgs/development/php-packages/composer/default.nix:8]: Why is this even done to begin with? The original URL should be the official upstream source, right?

Comment (drupol) [reply pkgs/development/php-packages/composer/default.nix]: Yeah I think it was something to do, but @SuperSandro2000 is right, there is no need to use those placeholders in the URL.

File: pkgs/development/php-packages/composer/default.nix
Comment (talyz) [inline pkgs/development/php-packages/composer/default.nix:8]: This isn't needed. The bot already updates composer. We should fetch from upstream's preferred source unless there are issues with it.

Comment (drupol) [reply pkgs/development/php-packages/composer/default.nix:8]: Indeed, this might not be mandatory, but wouldn't it be better to get everything from Github ?

**Comment by drupol** [comment]
Closing this issue by lack of interest and different point of view.

**Pull request closed without merging**
